osquery/core/system.cpp

```diff
@@ -71,6 +71,11 @@ FLAG(string,
      "hostname",
      "Field used to identify the host running osquery (hostname, uuid)");
 
+FLAG(bool,
+     disable_host_id_cache,
```

Comment:
The only downside I see is that on any platform where we can't retrieve the hardware system UUID, the default behavior will change so that every start of osquery will return a new UUID, unless osquery is invoked with a new option to cache the host UUID. 

I am not sure which platforms, if any, do not return the host UUID as understood by the OS - however, the code (system.cpp::generateHostUUID) handles this situation. 

I suspect it is just defensive coding (ie., generating an ephemeral UUID if unable to retrieve the host UUID). So, I think it's safe to switch the default behavior. I have made the change, but happy to revert again if someone says this will cause a backward compatibility issue.